Add a note that starting pserve with --daemon is no longer supported

prcutler opened the pull request:
For Pyramid 1.9 and newer, you cannot start Pyramid with pserve
--daemon.  This adds a note to the nginx deployment page to call
this out.

When I deployed my first Pyramid app this past August, the developers in IRC were kind enough to let me know that you can no longer start serve with --daemon.  This adds a note to the documentation to call this out.

I don't know if it should be expanded (for Linux users) to use a systemd service instead, but that could be added / expanded to the docs.

I did successfully build this in Sphinx and it looks correct - this is my first PR to a Pylons project, please share any feedback if there is something I can do different / better.

Thanks!

stevepiercy commented on docs/deployment/nginx.rst:
```diff
@@ -220,6 +220,10 @@ Running the pserve processes::
+.. note::
+
+   Starting the pserve process with ``--daemon`` is no longer supported in Pyramid 1.9 and newer.
+
```
@prcutler welcome to becoming a contributor!

I'm not sure if you know, the Cookbook is not official documentation but is community based. Where applicable, the Cookbook should refer to the official Pyramid (or whatever package) documentation. Sorry if this wasn't clear up front.

In this case, please remove the ``--daemon`` argument from the ``pserve`` launch commands, and update the new note to something like the following:

> Daemonization of pserve was [deprecated in Pyramid 1.6](https://docs.pylonsproject.org/projects/pyramid/en/latest/whatsnew-1.6.html#deprecations), then [removed in Pyramid 1.8](https://docs.pylonsproject.org/projects/pyramid/en/latest/whatsnew-1.8.html#backwards-incompatibilities).

prcutler commented:
Steve,

Thanks for the clarification and feedback.  I'll update with a new PR later today.

prcutler commented:
Thanks again for the feedback and I apologize for the delay in getting this updated - I've updated the PR and applied the feedback, removing the --daemon from the pserve command and changed the sentence to the recommendation with links.

It successfully built in Sphinx - please let me know if any additional changes are needed.

stevepiercy commented:
Thanks @prcutler!
